File: proselint/checks/misc/debased.py
"""
Debased language.

---
layout:     post
source:     ???
source_url: ???
title:      yelling
date:       2014-06-10
categories: writing
---

Too much yelling.

"""
from __future__ import annotations

from proselint.checks import CheckResult, existence_check, registry

examples_pass = [
    "Smoke phrase with nothing flagged.",
]

examples_fail = [
    "This leaves much to be desired.",
]


def check(text: str) -> list[CheckResult]:
    """Check the text."""
    err = "misc.debased"
    msg = "Debased language is a continuous temptation."

    items = [
        "a not unjustifiable assumption",
        "leaves much to be desired",
        "would serve no purpose",
        "a consideration which we should do well to bear in mind",
    ]

    return existence_check(text, items, err, msg)


registry.register("misc.debased", check)

File: proselint-rs/crates/proselint-checks/src/misc/debased.rs
use proselint_registry::checks::{Check, types::*, Padding};

pub const EXAMPLES_PASS: &[&str] = &["Smoke phrase with nothing flagged."];
pub const EXAMPLES_FAIL: &[&str] = &["This leaves much to be desired."];

const CHECK: Check = Check {
	check_type: &Existence {
		items: &[
			"a not unjustifiable assumption",
			"leaves much to be desired",
			"would serve no purpose",
			"a consideration which we should do well to bear in mind",
		],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "misc.debased",
	msg: "Debased language is a continuous temptation.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
